src/index.ts

```diff
@@ -47,10 +47,14 @@ client.login(BOT_TOKEN).catch((e: Error) => {
 
 const checkRights = (message: Message, rights: any): boolean => {
   const user = message.member;
-  rights.forEach(right => {
-    if (user.permissions.has(right)) return true;
-  });
-  return false;
+  const res = _.reduce(
+    rights,
+    (cur, right) => {
+      return cur || user.permissions.has(right);
+    },
+    false
+  );
+  return res;
```

Comment:
Well okay, just tried to use more fancy functions? :D
Nevertheless the execution time will be slower I guess. Since reduce will always iterate over all users while the previous implementation can return earlier. But I think we won't have that much users to notice any impact here.

Reply:
It iterates in both cases over all rights of the user, who is making a request. I don't expect a performance difference here. 
You are totally correct, that this is a change, just to be more fancy 😄

Reply:
I believe the previous implementation had a **worst case** runtime of O(n) since it can return and end the for each loop (at least e.g. java would do that). 
On the other hand the reduce **always** has O(n) runtime.